tests/docker_cluster.py

```diff
@@ -190,8 +206,9 @@ def _create_and_start_containers(self, master_image, slave_image=None, cmd=None,
             self.master,
             hostname=self.internal_master,
             cmd=cmd,
-            links=zip(self.slaves, self.slaves),
             **kwargs)
+
+        self._get_network().connect(self.master)
```

Comment:
master should be already connected, see line 204 where you call `_create_container`